### ISSUE TITLE
Several hundred treats edges missing from DrugCentral ingest

### ISSUE THREAD

**Issue by marcello-deluca** [RobokopU24/ORION#261]
Hi guys, 

The drugcentral ingest (https://github.com/RobokopU24/ORION/blob/master/parsers/drugcentral/src/loaddrugcentral.py) contains the following prompt for treats edge extraction:

`self.chemical_phenotype_query = 'select struct_id, relationship_name, umls_cui from public.omop_relationship ' \
                                        'where umls_cui is not null'`

However, after discussing with Tudor on Wednesday, there are around 600 treats edges that do not adequately map to a UMLS CUI. If we do not omit those and find a decent way to ontologically map them, we can increase our unique disease treats edges from this resource by almost 40% in one shot.

**Comment by eKathleenCarter** [comment]
Hey @marcello-deluca have you looked at this any more? 

I am looking at the omop_relationship table and for the umls_cui that are null, so are the snomed concept ids. There are about 3800 instances in the table. Did you find a good way to map these ontologically? The only identifier I see is a name

**Comment by marcello-deluca** [comment]
@eKathleenCarter the LLM based name resolving that I mentioned here operates on human-readable strings and seems to have very high accuracy / resolution for drug concepts

**Comment by eKathleenCarter** [comment]
Marcello, the problem would be that every time we build robokop it would need to hit your LLM based name resolver. Do you have your name resolver stood up as an API that we could do that? Otherwise we would have to manually map the 3000 terms (or more) each time.

**Comment by EvanDietzMorris** [comment]
The short answer is almost always: if we don't have real identifiers we don't want to include them. In this case, because it sounds like Marcello has put a lot of effort into looking into these specific mappings and thinks they are accurate, we might want to use them, but we wouldn't want to call a LLM service in parsing, especially if it's not a mature (and free) service we can rely on. We'd want to make a mapping file specific for drugs or drugcentral and include it in ORION. Let's discuss with @cbizon though.

**Comment by eKathleenCarter** [comment]
@marcello-deluca , do you have a mapping file for these instances? Would you like to make one with your LLM?

**Comment by cbizon** [comment]
Do we have some examples of these?

**Comment by eKathleenCarter** [comment]
examples:

![Image](https://github.com/user-attachments/assets/029be281-86d0-40cc-942e-90562bb2d271)


Full List: 

[omop_relationship_202503191541.csv](https://github.com/user-attachments/files/19449794/omop_relationship_202503191541.csv)

**Comment by eKathleenCarter** [comment]
@marcello-deluca what if we just ingest your drug-disease list? Would this fix most of this issue?

**Comment by cbizon** [comment]
@marcello-deluca do you think that Tudor would include these identifiers in drugcentral?  Is there any chance of just fixing the upstream data?

**Comment by marcello-deluca** [comment]
Tudor told me that the drugs do not map to a UMLS CUI so they cannot be mapped; this fell off my radar, I'll try to generate some IDs from this list and drop it on this issue later this week.

**Comment by cbizon** [comment]
> Tudor told me that the drugs do not map to a UMLS CUI so they cannot be mapped;

> I'll try to generate some IDs from this list

Can you explain the gap between these comments? :)

**Comment by marcello-deluca** [comment]
Generate IDs from this list using NameResPlus (They will not be UMLS)

**Comment by cbizon** [comment]
So is the problem that we can get identifiers, but that the identifiers are not UMLS so they don't fit into drug central's modeling?

**Comment by marcello-deluca** [comment]
Yes, the current Drug Central ingest does not ingest these indications into ROBOKOP because it uses the prompt self.chemical_phenotype_query = 'select struct_id, relationship_name, umls_cui from public.omop_relationship ' \ 'where umls_cui is not null'

**Comment by cbizon** [comment]
I guess what I'm getting at is - if you gave them a list of e.g. Pubchem identifiers, would they / could they figure out a way to include them?
